git-edit-index

```diff
@@ -194,12 +194,14 @@ def edit_index(index):
     """
     # We need to use a temporary file to store the current index and show it to
     # the user in an editor.
-    with tempfile.NamedTemporaryFile(mode='r+', prefix='git-edit-index-') as f:
+    tmp = tempfile.mkstemp(prefix='git-edit-index-')
```

Comment:
Instead of storing the result into `tmp` and using `tmp[0]` and `tmp[1]` (I bet any reader of the code will wonder what they are ;-)), I would use tuple unpacking to store the result into two variables and use these:
```python
tmp_fd, tmp_path = tempfile.mkstemp(prefix='git-edit-index-')
```